seismicpro/gather/gather.py

```diff
@@ -1490,11 +1551,37 @@ def _finalize_plot(self, ax, title, divider, event_headers, top_header,
 
         top_ax.set_title(**{'label': None, **title})
 
+        if len(ax.get_legend_handles_labels()[0]):
+            ax.legend()
+
+    def _process_masks(self, masks):
+        colors_iterator = cycle(['tab:red', 'tab:blue', 'tab:orange', 'tab:green', 'tab:purple', 'tab:pink',
+                                 'tab:olive', 'tab:cyan'])
+        masks_list = self._parse_headers_kwargs(masks, "masks")
+        for ix, mask_dict in enumerate(masks_list):
```

Comment:
```suggestion
        for ix, (mask_dict, default_color) in enumerate(zip(masks_list, colors_iterator)):
```
And remove next below